### ISSUE TITLE
Compile librelp on solaris

### ISSUE THREAD

**Issue by janorn** [rsyslog/librelp#5]
Doesn't look like it is possible to build relp on solaris.

Configure:

```
:~/tmp/librelp-1.2.7$ ./configure --disable-static --disable-tls
checking for a BSD-compatible install... /usr/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
...
*****************************************************
librelp will be compiled with the following settings:

Debug mode enabled:            no
TLS enabled:                   no
TLS authentication supported:  
```

Make:

```
:~/tmp/librelp-1.2.7$ gmake
gmake  all-recursive
gmake[1]: Entering directory `/data/pkgbuild/tmp/librelp-1.2.7'
Making all in src
gmake[2]: Entering directory `/data/pkgbuild/tmp/librelp-1.2.7/src'
  CC       librelp_la-relp.lo
relp.c: In function ‘engineEventLoopRun’:
relp.c:795:13: warning: unused variable ‘localRet’
relp.c: At top level:
relp.c:73:1: warning: ‘relpEngine_strerror_r’ defined but not used
  CC       librelp_la-relpsess.lo
  CC       librelp_la-relpframe.lo
  CC       librelp_la-tcp.lo
tcp.c: In function ‘EnableKeepAlive’:
tcp.c:624:38: warning: unused parameter ‘pSrv’
tcp.c: In function ‘setCORKopt’:
tcp.c:1515:19: error: ‘SOL_TCP’ undeclared (first use in this function)
tcp.c:1515:19: note: each undeclared identifier is reported only once for each function it appears in
gmake[2]: *** [librelp_la-tcp.lo] Error 1
gmake[2]: Leaving directory `/data/pkgbuild/tmp/librelp-1.2.7/src'
gmake[1]: *** [all-recursive] Error 1
gmake[1]: Leaving directory `/data/pkgbuild/tmp/librelp-1.2.7'
gmake: *** [all] Error 2
```

**Comment by rgerhards** [comment]
I just merged #49, which looks like it addresses the same thing.

**Comment by janorn** [comment]
It sure does!
Thanks
